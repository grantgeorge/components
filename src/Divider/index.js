// @flow

/**
 *
 * Divider
 *
 */

import * as React from 'react'
import styled from 'styled-components'

export const StyledDivider = styled.div`
  border-top: 1px solid #d9dadc;
  display: block;
  line-height: 1px;
  margin: 1.5rem 0;
  position: relative;
  text-align: center;
`

export const DividerTitle = styled.strong`
  background: #ffffff;
  font-size: 0.8rem;
  letter-spacing: 1px;
  padding: 0 20px;
  text-transform: uppercase;
`

type Props = {
  children?: React.Node,
  className?: string,
  style?: {},
}

const Divider = ({ children, className, style }: Props) => {
  return (
    <StyledDivider className={className} style={style}>
      <DividerTitle>{children}</DividerTitle>
    </StyledDivider>
  )
}

export default Divider
