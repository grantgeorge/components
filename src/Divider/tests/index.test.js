import React from 'react'
import { shallow } from 'enzyme'

import Divider from '../index'

describe('<Divider />', () => {
  let renderedComponent

  beforeEach(() => {
    renderedComponent = shallow(<Divider>Child</Divider>)
  })

  it('should render properly', () => {
    expect(renderedComponent).toMatchSnapshot()
  })
})
