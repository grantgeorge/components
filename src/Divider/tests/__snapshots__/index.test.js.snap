// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> should render properly 1`] = `
<styled.div>
  <styled.strong>
    Child
  </styled.strong>
</styled.div>
`;
